test/unittests/commands/ingest_xmlupload/test_filechecker.py

```diff
@@ -0,0 +1,114 @@
+
+    def test_execute_error_protocol_nonexisting(self) -> None:
+        file_problems = FileProblems([Path("foo/bar.txt"), Path("spam/eggs.pdf")], [])
+        expected = (
+            "Some files referenced in the <bitstream> tags of your XML file cannot be uploaded to the server.\n"
+            "\n"
+            "The following files don't exist on your computer:\n"
+            " - foo/bar.txt\n"
+            " - spam/eggs.pdf"
+        )
+        res = file_problems.execute_error_protocol()
+        assert res == expected
+
+    def test_execute_error_protocol_unsupported(self) -> None:
+        file_problems = FileProblems([], [Path("testdata/bitstreams/test.txt"), Path("testdata/bitstreams/test.pdf")])
+        expected = (
+            "Some files referenced in the <bitstream> tags of your XML file cannot be uploaded to the server.\n"
+            "\n"
+            "The following files have unsupported extensions:\n"
+            " - testdata/bitstreams/test.txt\n"
+            " - testdata/bitstreams/test.pdf"
+        )
+        res = file_problems.execute_error_protocol()
+        assert res == expected
+
+    def test_execute_error_protocol_inexisting_and_unsupported(self) -> None:
+        file_problems = FileProblems(
+            [Path("foo/bar.txt"), Path("spam/eggs.pdf")],
+            [Path("testdata/bitstreams/test.txt"), Path("testdata/bitstreams/test.pdf")],
+        )
+        expected = (
+            "Some files referenced in the <bitstream> tags of your XML file cannot be uploaded to the server.\n"
+            "\n"
+            "The following files don't exist on your computer:\n"
+            " - foo/bar.txt\n"
+            " - spam/eggs.pdf\n"
+            "\n"
+            "The following files have unsupported extensions:\n"
+            " - testdata/bitstreams/test.txt\n"
+            " - testdata/bitstreams/test.pdf"
+        )
+        res = file_problems.execute_error_protocol()
+        assert res == expected
+
+    def test_execute_error_protocol_with_file_saving(self) -> None:
+        file_problems = FileProblems(
+            [Path("foo/bar.txt"), Path("spam/eggs.pdf")], [Path("testdata/bitstreams/test.pdf")], maximum_prints=2
+        )
+        expected = (
+            "Some files referenced in the <bitstream> tags of your XML file cannot be uploaded to the server. "
+            "The full list of files with problems has been saved to 'file_problems.csv'."
+        )
+        res = file_problems.execute_error_protocol()
+        assert res == expected
+
+        output_file = Path("file_problems.csv")
+        df = pd.read_csv(output_file)
+
+        data_expected = {
+            "Files that don't exist on your computer": ["foo/bar.txt", "spam/eggs.pdf"],
+            "Files with unsupported extensions": ["testdata/bitstreams/test.pdf", None],
+        }
+        df_expected = pd.DataFrame(data_expected)
+        assert df.equals(df_expected)
+        output_file.unlink()
```

Comment:
see my comment above - I don't like the format of this report, and if you change that, the test will have to change too.